tests/Query/DateParser.test.ts

```diff
@@ -144,3 +144,64 @@ describe('Date Parser - correct delta for next & last month & quarter (Today is
         // because the length of Q2 in days is same as Q3.
     });
 });
+
+describe('DateParser - specific date ranges', () => {
+    it('should return dates for specific ranges', () => {
+        // Week (53 in a year)
+        testParsingDateRange('2020-W53', '2020-12-28', '2021-01-03');
+        testParsingDateRange('2021-W01', '2021-01-04', '2021-01-10');
+
+        // Week (52 in a year)
+        testParsingDateRange('2018-W52', '2018-12-24', '2018-12-30');
+        testParsingDateRange('2019-W01', '2018-12-31', '2019-01-06');
+        testParsingDateRange('2019-W02', '2019-01-07', '2019-01-13');
+
+        // Month
+        testParsingDateRange('2002-02', '2002-02-01', '2002-02-28');
+        testParsingDateRange('2013-06', '2013-06-01', '2013-06-30');
+        testParsingDateRange('2017-12', '2017-12-01', '2017-12-31');
+        testParsingDateRange('2024-02', '2024-02-01', '2024-02-29');
+
+        // Quarter
+        testParsingDateRange('2007-Q1', '2007-01-01', '2007-03-31');
+
+        // Year
+        testParsingDateRange('1996', '1996-01-01', '1996-12-31');
+        testParsingDateRange('2022', '2022-01-01', '2022-12-31');
```

Comment:
I asume all these numbers are 1-based... But what happens if zeroes are passed in the them.

Like `W00` - or what about `W0` or `W2`?

And `2022-00` ?

Reply:
They are below in the `it('should return invalid dates for erroneous specific ranges', ...`. I've added the `2022-00` case though =)

And just to note `0000` is a perfectly valid year =)

Pls resolve this conversation if this ok for you.